docs/data/tree-view/simple-tree-view/customization/IconExpansionTreeView.js

```diff
@@ -1,56 +1,27 @@
 import * as React from 'react';
 import Box from '@mui/material/Box';
 import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
-import { useTreeItem2Utils } from '@mui/x-tree-view/hooks';
-
 import { TreeItem2 } from '@mui/x-tree-view/TreeItem2';
 
-const CustomTreeItem = React.forwardRef(function MyTreeItem(props, ref) {
-  const { interactions } = useTreeItem2Utils({
-    itemId: props.itemId,
-    children: props.children,
-  });
-
-  const handleContentClick = (event) => {
-    event.defaultMuiPrevented = true;
-    interactions.handleSelection(event);
-  };
-
-  const handleIconContainerClick = (event) => {
-    interactions.handleExpansion(event);
-  };
-
-  return (
-    <TreeItem2
-      {...props}
-      ref={ref}
-      slotProps={{
-        content: { onClick: handleContentClick },
-        iconContainer: { onClick: handleIconContainerClick },
-      }}
-    />
-  );
-});
-
 export default function IconExpansionTreeView() {
   return (
     <Box sx={{ minHeight: 352, minWidth: 250 }}>
-      <SimpleTreeView aria-label="icon expansion">
-        <CustomTreeItem itemId="grid" label="Data Grid">
-          <CustomTreeItem itemId="grid-community" label="@mui/x-data-grid" />
-          <CustomTreeItem itemId="grid-pro" label="@mui/x-data-grid-pro" />
-          <CustomTreeItem itemId="grid-premium" label="@mui/x-data-grid-premium" />
-        </CustomTreeItem>
-        <CustomTreeItem itemId="pickers" label="Date and Time Pickers">
-          <CustomTreeItem itemId="pickers-community" label="@mui/x-date-pickers" />
-          <CustomTreeItem itemId="pickers-pro" label="@mui/x-date-pickers-pro" />
-        </CustomTreeItem>
-        <CustomTreeItem itemId="charts" label="Charts">
-          <CustomTreeItem itemId="charts-community" label="@mui/x-charts" />
-        </CustomTreeItem>
-        <CustomTreeItem itemId="tree-view" label="Tree View">
-          <CustomTreeItem itemId="tree-view-community" label="@mui/x-tree-view" />
-        </CustomTreeItem>
+      <SimpleTreeView aria-label="icon expansion" expansionTrigger="iconContainer">
+        <TreeItem2 itemId="grid" label="Data Grid">
```

Comment:
We tend to use `TreeItem` whenever both components are compatible.
That way people don't have any migration to do on this aspect for the next major.